devimint/src/tests.rs

```diff
@@ -1884,6 +1884,214 @@ pub async fn test_offline_client_initialization(
+        .await;
+
+    info!(target: LOG_DEVIMINT, "Modifying server configurations to add new meta module...");
+    modify_server_configs(&config_dir, &peer_ids).await?;
+
+    // Restart all servers with modified configs
+    info!(target: LOG_DEVIMINT, "Restarting all servers with modified configurations...");
+    for peer_id in peer_ids {
+        fed.start_server(process_mgr, peer_id.to_usize()).await?;
+    }
+
+    // Wait for federation to stabilize
+    info!(target: LOG_DEVIMINT, "Wait for peers to get back up");
+    fed.await_all_peers().await?;
+
+    // Use fedimint-cli dev wait to let the client read the new config in background
+    info!(target: LOG_DEVIMINT, "Waiting for client to fetch updated configuration...");
+    cmd!(client, "dev", "wait", "3")
+        .run()
+        .await
+        .context("Failed to wait for client config update")?;
+
+    // Test that client can detect the configuration change
+    info!(target: LOG_DEVIMINT, "Testing client detection of configuration changes...");
+    let updated_config = cmd!(client, "config")
+        .out_json()
+        .await
+        .context("Failed to get updated client config")?;
+
+    info!(target: LOG_DEVIMINT, "Updated config modules: {:?}", updated_config["modules"].as_object().unwrap().keys().collect::<Vec<_>>());
+
+    // Verify that the configuration has changed (new meta module should be present)
+    let initial_modules = initial_config["modules"].as_object().unwrap();
+    let updated_modules = updated_config["modules"].as_object().unwrap();
+
+    anyhow::ensure!(
+        updated_modules.len() > initial_modules.len(),
+        "Expected more modules in updated config. Initial: {}, Updated: {}",
+        initial_modules.len(),
+        updated_modules.len()
+    );
+
+    // Check if a new meta module was added
+    let mut found_new_meta = false;
+    for (module_id, module_config) in updated_modules {
+        if module_config["kind"].as_str() == Some("meta")
+            && !initial_modules.contains_key(module_id)
+        {
+            found_new_meta = true;
+            info!(target: LOG_DEVIMINT, "Found new meta module with id: {}", module_id);
+            break;
+        }
+    }
+
+    anyhow::ensure!(
+        found_new_meta,
+        "Expected to find new meta module in updated configuration"
+    );
+
+    // Verify client operations still work with the new configuration
+    info!(target: LOG_DEVIMINT, "Verifying client operations work with new configuration...");
+    let final_info = cmd!(client, "info")
+        .out_json()
+        .await
+        .context("Client info command failed with updated configuration")?;
+
+    info!(target: LOG_DEVIMINT, "Client successfully adapted to configuration changes: {:?}", final_info["federation_id"]);
+
+    Ok(())
+}
+
+/// Modify server configuration files to add a new meta module instance
+async fn modify_server_configs(config_dir: &Path, peer_ids: &[PeerId]) -> Result<()> {
```

Comment:
TIL: you don't have to patch the private or local config if the module doesn't have any

Reply:
?